backend/src/main/java/com/rota/api/StaffService.java

```diff
@@ -0,0 +1,51 @@
+package com.rota.api;
+
+import com.rota.api.dto.EngagementDto;
+import com.rota.database.orm.engagement.EngagementRepository;
+import com.rota.database.orm.staff.Staff;
+import com.rota.database.orm.staff.StaffRepository;
+import java.time.Instant;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class StaffService {
+  @Autowired
+  EngagementRepository engagementRepository;
+
+  @Autowired
+  StaffRepository staffRepository;
+
+  /**
+   * Returns staff member's engagements between start and end dates.
+   * Specifically engagements that start after the start date and end before the end date.
+   * Both start and end dates are optional and can be left as null.
+   *
+   * @param staffId Staff member's ID
+   * @param start start date or null to get from the beginning
+   * @param end end date
+   * @return A list of member's engagements
+   */
+  public List<EngagementDto> getStaffEngagementsBetween(int staffId, Instant start, Instant end) {
+    Instant startTime = start == null ? Instant.MIN : start;
+    Instant endTime = end == null ? Instant.MAX : end;
+    return engagementRepository.findByStaffId(staffId).stream()
+        .filter(engagement ->
+            engagement.getStart().isAfter(startTime)
```

Comment:
Should we make this 'after or equal to startTime', etc?

Reply:
I've not actually made these changes in my PR. But I think it'll be a good spot to change them over since that PR is already merged. I'll also change the Before one too.

@tjanowsk would you agree? or is it purposefully as only after/before.

Reply:
I don't mind, for continuous variables like time there's not much difference between greater and greater or equal.

Reply:
If we say 'show me everything from Monday 9am' we want all shifts that start at 9am on monday so I'd be inclusive here